lib/run-lifecycle.js

```diff
@@ -0,0 +1,28 @@
+const chalk = require('chalk')
+const checkpoint = require('./checkpoint')
+const figures = require('figures')
+const runExec = require('./run-exec')
+
+/**
+ * @name runLifecycle
+ * @param {object} ctx - all available context for a hook
+ * @param {object} ctx.args - merged program options
+ * @param {string} ctx.hookName - the ID of the hook
+ * @param {string} ctx.version - available version string
+ * @param {string?} [ctx.newVersion] - additional version string if available
+ */
+module.exports = async function (ctx) {
+  const { args, hookName } = ctx
+  const scripts = args.scripts
+  if (!scripts || !scripts[hookName]) return Promise.resolve()
+  const command = scripts[hookName]
+  if (typeof command === 'function') {
+    checkpoint(args, 'Running lifecycle script "%s"', [hookName])
+    checkpoint(args, '- execute command: "%s"', [command.name], chalk.blue(figures.info))
+    return command(ctx)
```

Comment:
I really like the idea of passing the full context to the handler(s), I want to suggest a slight change to the shape of this parameter that I think will help structure the data being consumed by lifecycles and make it a little more obvious where/how to add additional properties on our end (without conflict):

```js
runLifecycle(name: string, { args: object, context: object })
// or
runLifecycle(payload = {name: string, args: object, context: object })
```